src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1496,7 +1496,7 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
 
                 case SymbolKind.Alias:
                     {
-                        var alias = symbol as AliasSymbol;
+                        var alias = (AliasSymbol)symbol;
                         symbol = alias.Target;
```

Comment:
To clarify, the pattern of using "as" operator for casts that aren't expected to fail is forbidden in compilers because it negatively affect readability of the code. It also makes it harder to find bugs around null reference dereference before the code is submitted. We discussed this at length and got  to an agreement that a developer should be explicit in the assumptions he/she is making. I.e. if the assumption is "the instance must be of that type", then a regular cast should be used, which also has a benefit of getting InvalidCastException when the assumption is incorrect. If the assumption is "the instance might be of that type", then "as" cast should be used, followed by a null check before the result is dereferenced. There was some preexisting code in the codebase that didn't follow the rule, so we might still run into violations like this.